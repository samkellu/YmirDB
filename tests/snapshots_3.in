LIST KEYS
LIST ENTRIES
LIST SNAPSHOTS
SET d 4
SET c 3 d
SET b 2 c
SET a 1 b
LIST KEYS
LIST ENTRIES
LIST SNAPSHOTS
TYPE a
TYPE b
TYPE c
TYPE d
LEN a
LEN b
LEN c
LEN d
FORWARD a
FORWARD b
FORWARD c
FORWARD d
BACKWARD a
BACKWARD b
BACKWARD c
BACKWARD d
SNAPSHOT
APPEND d -1
GET d
SORT d
GET d
SET e 0
APPEND d e
LIST KEYS
LIST ENTRIES
LIST SNAPSHOTS
TYPE a
TYPE b
TYPE c
TYPE d
TYPE e
LEN a
LEN b
LEN c
LEN d
LEN e
FORWARD a
FORWARD b
FORWARD c
FORWARD d
FORWARD e
BACKWARD a
BACKWARD b
BACKWARD c
BACKWARD d
BACKWARD e
SNAPSHOT
CHECKOUT 1
