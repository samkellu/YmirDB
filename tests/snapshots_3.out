> no keys

> no entries

> no snapshots

> ok

> ok

> ok

> ok

> a
b
c
d

> a [1 b]
b [2 c]
c [3 d]
d [4]

> no snapshots

> general

> general

> general

> simple

> 4

> 3

> 2

> 1

> b, c, d

> c, d

> d

> nil

> nil

> a

> a, b

> a, b, c

> saved as snapshot 1

> ok

> [4 -1]

> ok

> [-1 4]

> ok

> ok

> e
a
b
c
d

> e [0]
a [1 b]
b [2 c]
c [3 d]
d [-1 4 e]

=================================================================
==13886==ERROR: AddressSanitizer: attempting free on address which was not malloc()-ed: 0x613000000160 in thread T0
    #0 0x7f041513d7cf in __interceptor_free (/lib/x86_64-linux-gnu/libasan.so.5+0x10d7cf)
    #1 0x7f0415ab8a09 in command_checkout /mnt/c/users/samke/desktop/uni/comp2017/ymirdb/ymirdb.c:521
    #2 0x7f0415abce80 in main /mnt/c/users/samke/desktop/uni/comp2017/ymirdb/ymirdb.c:952
    #3 0x7f0414e540b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x240b2)
    #4 0x7f0415ab342d in _start (/mnt/c/Users/samke/desktop/uni/comp2017/ymirdb/ymirdb+0x342d)

0x613000000160 is located 288 bytes inside of 360-byte region [0x613000000040,0x6130000001a8)
allocated by thread T0 here:
    #0 0x7f041513dffe in __interceptor_realloc (/lib/x86_64-linux-gnu/libasan.so.5+0x10dffe)
    #1 0x7f0415ab52d8 in command_set /mnt/c/users/samke/desktop/uni/comp2017/ymirdb/ymirdb.c:252
    #2 0x7f0415abcb1d in main /mnt/c/users/samke/desktop/uni/comp2017/ymirdb/ymirdb.c:936
    #3 0x7f0414e540b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x240b2)

SUMMARY: AddressSanitizer: bad-free (/lib/x86_64-linux-gnu/libasan.so.5+0x10d7cf) in __interceptor_free
==13886==ABORTING
